src/components/c-google-maps.vue

```diff
@@ -0,0 +1,372 @@
+         */
+        boundingTimeout: null,
+
+        /**
+         * @type {Boolean} Determines if the component is allowed to auto update the bounding.
+         */
+        allowAutoUpdates: true,
+      };
+    },
+
+    computed: {
+      ...mapGetters('session', [
+        'getGoogleMapsApiKey',
+      ]),
+
+      /**
+       * Maps the locations to the Google Maps required format.
+       *
+       * @returns {Array.<Object>}
+       */
+      mappedLocations() {
+        return this.locations?.map(location => ({
+          lat: parseFloat(location.lat),
+          lng: parseFloat(location.lng),
+          icon: location.icon,
+          geocode: location.geocode,
+          title: location.title,
+          referer: location, // Keep a reference to the original for event payloads.
+        })) || [];
+      },
+
+      /**
+       * Returns a style definition based on the given theme.
+       *
+       * @returns {Array.<Object>}
+       */
+      mapStyles() {
+        switch (this.theme) {
+          case 'gray':
+            return [...GOOGLE_MAPS_THEME_GRAY, ...this.styles];
+
+          default:
+            return this.styles;
+        }
+      },
+    },
+    // watch: {},
+
+    beforeCreate() {
+      const apiKey = store.getters['session/getGoogleMapsApiKey'];
+
+      if (!apiKey) {
+        throw new Error('No Google Maps API key provided.');
+      }
+
+      const url = `https://maps.googleapis.com/maps/api/js?key=${apiKey}&libraries=places`;
+
+      /**
+       * Callback for the load event of the map script.
+       */
+      const callback = () => {
+        geocoder = new window.google.maps.Geocoder();
+
+        this.$nextTick(() => {
+          this.createMapInstance();
+        });
+      };
+
+      loadScript(url, callback, { async: true });
+    },
+    // created() {},
+    // beforeMount() {},
+    mounted() {
+      this.$el.addEventListener('mousedown', this.disableAutoUpdates, { once: true, passive: true });
+    },
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    beforeDestroy() {
+      this.$el.removeEventListener('mousedown', this.disableAutoUpdates);
+    },
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Updates the 'allowAutoUpdates' flag.
+       */
+      disableAutoUpdates() {
+        this.allowAutoUpdates = false;
+      },
+
+      /**
+       * Initialize the google maps with the default values.
+       *
+       * TODO: note, that Google does not support destroying an instance... @see https://stackoverflow.com/questions/10485582/what-is-the-proper-way-to-destroy-a-map-instance
+       */
+      createMapInstance() {
+        if (!this.mappedLocations?.length && !this.center) {
+          const errorMsg = 'Neither locations nor a center coordinate was given.'
+            + ' At least one of them is needed to create a Google Maps.'; // eslint-disable-line vue/max-len
```

Comment:
suggestion: if you did add the line break, please remove it. It makes the string harder to maintain. It's OK to have strings that exceed the line length. If possible remove `// eslint-disable-line vue/max-len` (but not mandatory).